pyproject.toml

```diff
@@ -23,7 +23,7 @@ include = ["LICENSE"]
 packages = [{ include = "xrpl" }]
 
 [tool.poetry.dependencies]
-python = "^3.8"
+python = ">=3.8.1,<3.14"
```

Comment:
Why is there a max number here?

Reply:
One of the dependencies `darglint` has an upper bound on the supported Python version. Here is the error message:
```
The current project's supported Python range (>=3.8.1) is not compatible with some of the required packages Python requirement:
  - darglint requires Python >=3.6,<4.0, so it will not be satisfied for Python >=4.0
 ```

Reply:
Here's the complete error log:
```
➜  xrpl-py git:(dependabot/pip/black-24.8.0) poetry lock
Updating dependencies
Resolving dependencies... (2.1s)

The current project's supported Python range (>=3.8.1) is not compatible with some of the required packages Python requirement:
  - darglint requires Python >=3.6,<4.0, so it will not be satisfied for Python >=4.0
  - darglint requires Python >=3.6,<4.0, so it will not be satisfied for Python >=4.0
  - darglint requires Python >=3.6,<4.0, so it will not be satisfied for Python >=4.0
  - darglint requires Python >=3.6,<4.0, so it will not be satisfied for Python >=4.0
  - darglint requires Python >=3.6,<4.0, so it will not be satisfied for Python >=4.0

Because no versions of darglint match >1.5.8,<1.6.0 || >1.6.0,<1.7.0 || >1.7.0,<1.8.0 || >1.8.0,<1.8.1 || >1.8.1,<2.0.0
 and darglint (1.5.8) requires Python >=3.6,<4.0, darglint is forbidden.
And because darglint (1.6.0) requires Python >=3.6,<4.0, darglint is forbidden.
And because darglint (1.7.0) requires Python >=3.6,<4.0
 and darglint (1.8.0) requires Python >=3.6,<4.0, darglint is forbidden.
So, because darglint (1.8.1) requires Python >=3.6,<4.0
 and xrpl-py depends on darglint (^1.5.8), version solving failed.
```

Reply:
I have a PR replacing darglint, so this is probably a moot point: #749 

It's a very outdated package.

Reply:
This can include `<4.0`, but it shouldn't be gating minor releases. Most of them work automatically (see #753), so gating it would actually be worse/require more work, since this would prevent it from working automatically without a version bump.

Reply:
fixed in 43fc8755

Reply:
I haven't yet looked at the darglint PR. We could revisit the gating upper bound later.